setup.py

```diff
@@ -0,0 +1,55 @@
+# Available at setup time due to pyproject.toml
+from pybind11.setup_helpers import Pybind11Extension, build_ext
+from Cython.Build import cythonize
+from setuptools import setup, Extension
+import pathlib
+
+__version__ = "0.0.1"
+
+# The main interface is through Pybind11Extension.
+# * You can add cxx_std=11/14/17, and then build_ext can be removed.
+# * You can set include_pybind11=false to add the include directory yourself,
+#   say from a submodule.
+#
+# Note:
+#   Sort input source files if you glob sources to ensure bit-for-bit
+#   reproducible builds (https://github.com/pybind/python_example/pull/53)
+
+# ext_modules = [
+#     Pybind11Extension(
+#         "Knap",
+#         ["cpp/KnapsackUtilsCpp.cpp"],
+#         # Example: passing in the version to the compiled code
+#         define_macros=[("VERSION_INFO", __version__)],
+#     ),
+# ]
+
+# setup(
+#     name="KnapsackUtilsCpp",
+#     version=__version__,
+#     author="Michał Radwański (enedil)",
+#     author_email="enedil (at) outlook.com",
+#     description="KnapsackUtils",
+#     long_description="",
+#     ext_modules=ext_modules,
+#     extras_require={"test": "pytest"},
+#     # Currently, build_ext only provides an optional "highest supported C++
+#     # level" feature, but in the future it may provide more features.
+#     cmdclass={"build_ext": build_ext},
+#     zip_safe=False,
+#     python_requires=">=3.7",
+# )
+
+extensions = [
```

Comment:
I assume I'll need to hack all this stuff for windows? also your include dirs should prob just be relative to the dependency folder or whatever, why reffing a user folder?